pymatgen/io/openmm/tests/test_generators.py

```diff
@@ -40,8 +44,18 @@ def test_get_input_set(self):
         }
         assert set(input_set.settings["atom_resnames"]) == {"CCO", "H2O"}
         assert len(input_set.settings["atom_types"]) == 780
+        monty.serialization.dumpfn(input_set, "./test")
         assert input_set.validate()
 
+    def test_dump_load_input_set(self):
+        input_set = OpenMMSet.from_directory(input_set_path)
+        with tempfile.TemporaryDirectory() as tmpdir:
+
+            monty.serialization.dumpfn(input_set, tmpdir + "/input_set.json")
+            input_set2 = monty.serialization.loadfn(tmpdir + "/input_set.json")
+            # TODO: currently failing equality test due to bug
+            assert input_set == input_set2
+
```

Comment:
Hey @rkingsbury. For some reason testing equality on InputSets is failing. The issue is that when `InputSet.__getitem__(key)` is being called by collections.abc, it is being passed a `(key, value)` tuple instead of just the `key`. I was able to reproduce the issue with the PackmolSet by calling `packmol_set == packmol_set`, so this is an issue beyond just the OpenMM IO I wrote and may have to be addressed in `pymatgen.io.core`. Just wanted to give you a heads up, I can raise an issue on `pymatgen` too if desired.

Reply:
Thanks @orioncohen ; must be a flaw in `__getitem__` . Feel free to open an issue or a PR if you see a clear resolution. I'll try to prioritize addressing that asap